src/core/connect/dbus_connect.cpp

```diff
@@ -0,0 +1,281 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2022 YADRO
+
+#include "dbus_connect.hpp"
+
+#include <systemd/sd-bus.h>
+
+#include <core/connect/connect.hpp>
+#include <core/exceptions.hpp>
+#include <phosphor-logging/log.hpp>
+#include <sdbusplus/bus.hpp>
+#include <sdbusplus/bus/match.hpp>
+#include <sdbusplus/exception.hpp>
+
+#include <chrono>
+#include <cstring>
+#include <functional>
+#include <memory>
+
+namespace app
+{
+namespace connect
+{
+
+using namespace sdbusplus::bus;
+using namespace std::chrono;
+using namespace std::chrono_literals;
+using namespace std::literals;
+using namespace phosphor::logging;
+
+DBusConnect::DBusConnect() :
+    dbusCallWindow(DBusCallWindow::ready), alive(false), dbusConnect(nullptr)
+{}
+
+DBusConnect::~DBusConnect() noexcept
+{
+    this->terminate();
+    this->disconnect();
+}
+
+void DBusConnect::run()
+{
+    alive.store(true);
+    thread =
+        std::make_unique<std::thread>(std::bind(&DBusConnect::process, this));
```

Comment:
Why? Do you believe the `std::thread` might cause an exception on initialization? 
The `DBusConnect::process` hasn't public `throw` statements. All exceptions are handled internally.

Reply:
But as I can see the third party parts (e.g. fastcgi++/manager) used by this project starts the threads also without handling the exception.
Thus, It can be left as is  just for the uniformity.

Reply:
Ok, I'll leave it as is.